core/templates/domain/story/story-update.service.ts

```diff
@@ -216,9 +216,11 @@ angular.module('oppia').factory('StoryUpdateService', [
           title: nodeTitle
         }, function(changeDict, story) {
           // Apply.
+          StoryEditorStateService.setExpIdsChanged();
```

Comment:
Maybe do this after line 220 i.e, after adding a newNode? (It's fine if you want to leave this as-is)

Reply:
Done, also changed elsewhere in file.